examples/timeseries/timeseries_transformer_classification.py

```diff
@@ -0,0 +1,189 @@
+        self.attention_dropout = keras.layers.Dropout(dropout)
+        self.attention_norm = keras.layers.LayerNormalization(epsilon=1e-6)
+
+        self.ff_conv1 = keras.layers.Conv1D(
+            filters=ff_dim, kernel_size=1, activation='relu')
+
+        # self.ff_conv2 at build()
+        self.ff_dropout = keras.layers.Dropout(dropout)
+        self.ff_norm = keras.layers.LayerNormalization(epsilon=1e-6)
+
+    def build(self, input_shape):
+        self.ff_conv2 = keras.layers.Conv1D(
+            filters=input_shape[-1], kernel_size=1)
+
+    def call(self, inputs, training, **kwargs):
+        x = self.attention_norm(inputs, **kwargs)
+        x = self.attention(x, **kwargs)
+        x = self.attention_dropout(x, training=training, **kwargs)
+        res = x + inputs
+
+        x = self.ff_norm(res, **kwargs)
+        x = self.ff_conv1(x, **kwargs)
+        x = self.ff_dropout(x, training=training, **kwargs)
+        x = self.ff_conv2(x, **kwargs)
+        x = self.ff_dropout(x, training=training, **kwargs)
+        return x + res
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+"""
+Finally, we can wrap every layer up and add the ability to stack several of them together.
+"""
+
+class TransformerEncoder(keras.Model):
+    def __init__(self, name='TransformerEncoder', num_heads=1, head_size=64, ff_dim=None, num_layers=1, dropout=0, **kwargs):
+        super().__init__(name=name, **kwargs)
+        if ff_dim is None: # sensible default
+            ff_dim = head_size
+        self.dropout = dropout
+        self.attention_layers = [SelfAttentionBlock(
+            num_heads=num_heads, head_size=head_size, ff_dim=ff_dim, dropout=dropout) for _ in range(num_layers)]
+
+    def call(self, inputs, training, **kwargs):
+        x = inputs # (batch size, sequence length, features)
+        for attention_layer in self.attention_layers:
+            x = attention_layer(x, training, **kwargs)
+
+        return x # (batch size, sequence length, features)
+
+"""
+The main part of our model is now complete and we can now proceed to add the final Multi-Layer Perceptron classification "head".
+Apart from a stack of `Dense` layers, we need to reduce the output tensor of the `TransformerEncoder` part of our model,
+down to a vector of features for each data point in the current batch.
+A common way to achieve this is to use some sort of pooling layer.
+For this example, a `GlobalAveragePooling1D` layer is sufficient.
+Feel free to experiment with different pooling layers if you wish, encorporating domain expertise.
+"""
+
+def build_model(input_shape, n_classes, head_size, num_heads=4, num_layers=2, dropout=0.2, dropout_mlps=0.5, mlp=[],ff_dim=None, activation='relu', optimizer='adamW'):
+    model_input = keras.Input(shape=input_shape)
+    x = model_input
+    x = TransformerEncoder(num_heads=num_heads, head_size=head_size, ff_dim=None, num_layers=num_layers, dropout=dropout)(x)
```

Comment:
If you name the transformer encoder block `TransformerEncoder`, then you'd simply get:

```
for _ in range(N):
   x = TransformerEncoder(...)(x)
```